src/SFML/System/Android/SuspendAwareClock.hpp

```diff
@@ -22,40 +22,51 @@
 //
 ////////////////////////////////////////////////////////////
 
-#ifndef SFML_CLOCKIMPLUNIX_HPP
-#define SFML_CLOCKIMPLUNIX_HPP
+
+#ifndef SFML_SUSPENDAWARECLOCK_HPP
+#define SFML_SUSPENDAWARECLOCK_HPP
 
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
-#include <SFML/Config.hpp>
-#include <SFML/System/Time.hpp>
+#include <chrono>
+#include <type_traits>
 
+#ifdef CLOCK_BOOTTIME
 
-namespace sf
-{
-namespace priv
+namespace sf::priv
 {
 ////////////////////////////////////////////////////////////
-/// \brief Unix implementation of sf::Clock
+/// \brief Android, chrono-compatible, suspend-aware clock
+///
+/// Linux steady clock is represented by CLOCK_MONOTONIC.
+/// However, this implementation does not work properly for
+/// long-running clocks that work in background when the
+/// system is suspended.
+///
+/// SuspendAwareClock uses CLOCK_BOOTTIME which is identical
+/// to CLOCK_MONOTONIC, except that it also includes any time
+/// that the system is suspended.
+///
+/// In most cases, CLOCK_MONOTONIC is a better choice. Make
+/// sure this implementation is required for your use case.
 ///
 ////////////////////////////////////////////////////////////
-class ClockImpl
+class SuspendAwareClock
 {
 public:
+    using duration = std::chrono::nanoseconds;
+    using rep = duration::rep;
+    using period = duration::period;
+    using time_point = std::chrono::time_point<SuspendAwareClock, duration>;
```

Comment:
These types define the [`TrivialClock`](https://en.cppreference.com/w/cpp/named_req/TrivialClock) concept and is expected for standard-like clocks, correct?

Could you maybe add a comment to express this? Otherwise people might wonder about the naming convention.

Reply:
Correct, `TrivialClock` to be specific but more importantly it will pass the check for a regular `Clock` concept [here](https://en.cppreference.com/w/cpp/chrono/is_clock).

Yes, the naming inconsistency might startle some. I'll add comments.